pkg/deploy/registry/instantiate/rest.go

```diff
@@ -102,6 +102,8 @@ func (r *REST) Create(ctx apirequest.Context, obj runtime.Object) (runtime.Objec
 		return nil, err
 	}
 
+	// Do unconditional update here to guarantee it always succeed.
+	config.ResourceVersion = ""
 	updated, _, err := r.store.Update(ctx, config.Name, rest.DefaultUpdatedObjectInfo(config, kapi.Scheme))
```

Comment:
It seems like the entire method could be done in a `RetryOnConflict` block.  The end user wouldn't know that you're retrying for them, but the conditional check in the code would re-run so that changes that affect the branching path in here would be reflected.